examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,453 @@
+# augment the on the fly during training
+train_dataset = (
+    train_loader.shuffle(len(x_train)).map(train_preprocessing).batch(batch_size)
+)
+# only rescale
+test_dataset = (
+    test_loader.shuffle(len(x_test)).map(test_preprocessing).batch(batch_size)
+)
+
+"""
+Visualize an augmented CT scan
+"""
+
+import matplotlib.pyplot as plt
+
+data = train_dataset.take(1)
+images, labels = list(data)[0]
+images = images.numpy()
+image = images[0]
+print("Dimension of the CT scan is:", image.shape)
+plt.imshow(np.squeeze(image[:, :, 80]), cmap="gray")
+
+# visualize montage of slices
+# 10 rows and 10 columns for 100 slices of the CT scan
+plot_slices(10, 10, 128, 128, image[:, :, :100])
+
+"""
+## Define a 3D convolutional neural network
+
+To make the model easier to understand, we structure it into blocks.
+The architecture of the 3D CNN used in this example
+is based on [this paper](https://arxiv.org/abs/2007.13224).
+"""
+
+
+def get_model(width=128, height=128, depth=128):
+    """build a 3D convolutional neural network model"""
+
+    inputs = keras.Input((width, height, depth, 1))
+
+    x = layers.Conv3D(filters=64, kernel_size=3, activation="relu")(inputs)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=128, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=256, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=512, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=512, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.GlobalAveragePooling3D()(x)
+    x = layers.Dense(units=512, activation="relu")(x)
+    x = layers.Dropout(0.3)(x)
+
+    outputs = layers.Dense(units=1, activation="sigmoid")(x)
+
+    # define the model
+    model = keras.Model(inputs, outputs, name="3dcnn")
+    return model
+
+
+# build model
+model = get_model(width=128, height=128, depth=128)
+model.summary()
+
+"""
+## Train model
+"""
+
+# compile model
+initial_learning_rate = 0.0001
+lr_schedule = keras.optimizers.schedules.ExponentialDecay(
+    initial_learning_rate, decay_steps=100000, decay_rate=0.96, staircase=True
+)
+model.compile(
+    loss="binary_crossentropy",
+    optimizer=keras.optimizers.Adam(learning_rate=lr_schedule),
+    metrics=["acc"],
+)
+
+# define callbacks
+checkpoint_cb = keras.callbacks.ModelCheckpoint(
+    "3d_image_classification.h5", save_best_only=True
+)
+early_stopping_cb = keras.callbacks.EarlyStopping(monitor="val_acc", patience=10)
+
+# train the model, doing validation at the end of each epoch.
+epochs = 100
+model.fit(
+    train_dataset,
+    validation_data=test_dataset,
```

Comment:
If you're going to use it as validation dataset, please name it `validation_dataset` to avoid confusion.

Reply:
Named everything validation instead of test.